cloud/scope/cluster.go

```diff
@@ -269,6 +269,14 @@ func (s *ClusterScope) PrivateDNSSpec() *azure.PrivateDNSSpec {
 	return spec
 }
 
+// AvailabilitySetSpecs returns the availability set specs.
+func (s *ClusterScope) AvailabilitySetSpecs() []azure.AvailabilitySetSpec {
+	return []azure.AvailabilitySetSpec{{
+		Name:                    azure.GenerateAvailabilitySetName(s.ClusterName(), azure.ControlPlaneNodeGroup),
+		DesiredFaultDomainCount: azure.DefaultFaultDomainCount,
```

Comment:
Is the fault domain count always 3? I know you had questions about it before the break, did you answer those?

Reply:
Ah I spoke too soon... I see it's replaced later based on the sku cache return.

Reply:
Yeah, I checked the possible list of fault domains using `az vm list-skus --resource-type availabilitySets --query '[?name==`Aligned`].{Location:locationInfo[0].location, MaximumFaultDomainCount:capabilities[0].value}' -o Table
`, and the max was 3. So, I thought it makes sense to do a `min(fdCountOfSku, desiredFDCount)`